File: backend/app/Domain/Settings/Models/PaymentMethod.php
<?php

declare(strict_types=1);

namespace App\Domain\Settings\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

/**
 * Mode de paiement paramétrable.
 *
 * @property int $id
 * @property string $code
 * @property string $name
 * @property string $type
 * @property bool $is_active
 * @property int $position
 * @property Carbon|null $created_at
 * @property Carbon|null $updated_at
 */
final class PaymentMethod extends Model
{
    /** Types de règlement supportés. */
    public const TYPES = ['cash', 'cheque', 'transfer', 'card', 'other'];

    protected $fillable = ['code', 'name', 'type', 'is_active', 'position'];

    /**
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'position' => 'integer',
        ];
    }
}

File: mobile/lib/models/payment_method.dart
/// Mode de paiement (GET /payment-methods, PaymentMethodResource).
class PaymentMethod {
  const PaymentMethod({
    required this.id,
    required this.code,
    required this.name,
    this.type,
    this.isActive = true,
  });

  final int id;
  final String code;
  final String name;
  final String? type;
  final bool isActive;

  factory PaymentMethod.fromJson(Map<String, dynamic> json) => PaymentMethod(
        id: json['id'] as int,
        code: json['code'] as String? ?? '',
        name: json['name'] as String? ?? '',
        type: json['type'] as String?,
        isActive: json['is_active'] != false,
      );
}
